10a-walkthrough-6.Rmd

```diff
@@ -0,0 +1,677 @@
+    # Exclude outliers
+    filter(f < 500000) %>% 
+    # Compute male student to female student ratio 
+    mutate(ratio = m / f) %>% 
+    select(-c(age, disability))
+```
+
+```{r check factor sample sizes}
+model_data %>% count(year)
+```
+
+Earlier we asked the question Do we have enough data points for the count of female students to learn about the ratio of female to male students? Similarly, we should ask the question do we have enough data points across our year variable to learn about how the ratio of female to male students has changed over time. We can visualize the ratio values across all years to check. Note the use of `geom_jitter` to spread the points horizontally so we can estimate the quantities better: 
+
+```{r}
+ggplot(data = model_data, aes(x = year, y = ratio)) + 
+    geom_jitter(alpha = .5) + 
+    labs(title = "Male to female ratio across years (jittered)")
+```
+
+Each year seems to have data points that can be considered when we fit the model. This means that there are enough data points to help us learn how the year variable predicts the ratio variable. 
+
+We fit the linear regression model by passing the argument `ratio ~ year` to the function `lm`. In R, the `~` usually indicates a formula. In this case, the formula is the variable `year` as a predictor of the variable `ratio`. The final argument we pass to `lm` is `data = model_data`, which tells R to look for the variables `ratio` and `year` in the dataset `model_data`. The results of the model are called a "model object." We'll store the model object in `ratio_year`:
+
+```{r fit model for year predicting ratio }
+ratio_year <- lm(ratio ~ year, data = model_data)
+```
+
+Each model object is filled with all sorts of model information. We can look at this information using the fuction `summary`: 
+
+```{r model summary}
+summary(ratio_year)
+```
+
+Here's how we can interpret the `Estimate` column: The estimate of the `(Intercept)` is 2.03356, which is the estimated value of the `ratio` variable when the `year` variable is "2012". Note that the value `year2012` isn't present in the in the list of rownames. That's because the `(Intercept)` row represents `year2012`. In linear regression models that use factor variables as predictors, the first level of the factor is the intercept. Sometimes this level is called a "dummy variable". The remaining rows of the model output show how much each year differs from the intercept, 2012. Fot example, `year2013` has an estimate of -0.01205, which suggests that on average the value of `ratio` is .01205 less than 2.03356. On average, the ratio of `year2014` is .02372 less than 2.03356. The `t value` column tells us the size of difference between the estimated value of the ratio for each year and the estimated value of the ratio of the intercept. Generally speaking, the larger the t value, the larger the chance that any difference between the coefficient of a factor level and the intercept are significant. 
```

Comment:
thoughts on how we should round values? My first thought is to three decimal points (at least for most cases)